pandas_transform/pandas_transform.md

```diff
@@ -157,54 +147,64 @@ You might have noticed that in the above example, everything in the first column
 
 **Run this code** to print out `col1` from the DataFrame `df`.
 
-<div class="python_data">
-<lia-keep>
-<script type="text/x-sage">
-#print(df['col1'])
-</script>
-</lia-keep>
-</div>
+```python
+print(df['col1'])
+
+```
+@Pyodide.eval
 
 Did you notice the `Name` and `dtype` at the bottom? Those tell you the name of the series and its data type. Try changing `col1` to `col2` or `col3` to see what type of data those series contain.
 
 
 ### Quiz: `pandas` package
 
-@sage
-<div class="python">
-<lia-keep>
-<script type="text/x-sage">
+```python
 d = {'col1': [1, 5, 7], 'col2': [3, .4, -2], 'col3':["yes", "no","blue"]};
 df = pd.DataFrame(data=d);
 print(df)
-</script>
-</lia-keep>
+
+```
+@Pyodide.eval
+
+You show this code to a friend. They try to run it in a Jupyter notebook on their computer, but it doesn't run, despite being exactly what ran for you on the previous page! 
+
+This is an extremely common and frustrating experience, what are some possible reasons why it isn't working for them?
+
+[[ ]] The code shouldn't have worked for either of you.
+[[X]] You had imported a package that they don't yet have.
+[[X]] They have defined a variable or module differently than you did.
+[[ ]] Their computer doesn't like them.
+***
+<div class="answer">
+
+When working code is copied to another location and then fails to work, it is usually because of a missing package or a differently defined variable.
+
 </div>
+***
 
 
-Your friend's code (above) isn't running, which is extremely frustrating because they copied and pasted it from the previous page of this module. What line must they add to make it work?
+Your friend shows you the error message they got when running the code: `NameError: name 'pd' is not defined`. What command should they enter before the first line of code?
 
 [[import pandas as pd]]
-[[?]] Hint: Run the code. What does the error message say is wrong?
-[[?]] Hint: What do we need to include at the top in order to make `pandas` commands available to us?
 ***
 <div class = "answer">
 
 `NameError: name 'pd' is not defined` is a helpful error message because it tells you exactly what you friend forgot: they never defined `pd`!
```

Comment:
```suggestion
`NameError: name 'pd' is not defined` is a helpful error message because it tells you exactly what your friend forgot: they never defined `pd`!
```